From: Lasse Collin <lasse.collin@tukaani.org>
Subject: xz: cleanup CRC32 edits from 2018
Date: Wed, 20 Mar 2024 20:38:39 +0200

The commit faa16bc404d72a5a ("lib: Use existing define with polynomial")
in 2018 added a dependency on <linux/crc32poly.h> to avoid duplicating the
same constant in multiple files.  Two months later it was found to be a
bad idea and the commit 242cdad873a75652 ("lib/xz: Put CRC32_POLY_LE in
xz_private.h") added the definition of CRC32_POLY_LE macro into
xz_private.h to avoid including <linux/crc32poly.h>.

xz_private.h is a wrong place for it too.  Revert back to the upstream
version which has the poly in xz_crc32_init() in xz_crc32.c.

Link: https://lkml.kernel.org/r/20240320183846.19475-7-lasse.collin@tukaani.org
Signed-off-by: Lasse Collin <lasse.collin@tukaani.org>
Reviewed-by: Jia Tan <jiat0218@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/xz/xz_crc32.c   |    2 +-
 lib/xz/xz_private.h |    4 ----
 2 files changed, 1 insertion(+), 5 deletions(-)

--- a/lib/xz/xz_crc32.c~xz-cleanup-crc32-edits-from-2018
+++ a/lib/xz/xz_crc32.c
@@ -28,7 +28,7 @@ STATIC_RW_DATA uint32_t xz_crc32_table[2
 
 XZ_EXTERN void xz_crc32_init(void)
 {
-	const uint32_t poly = CRC32_POLY_LE;
+	const uint32_t poly = 0xEDB88320;
 
 	uint32_t i;
 	uint32_t j;
--- a/lib/xz/xz_private.h~xz-cleanup-crc32-edits-from-2018
+++ a/lib/xz/xz_private.h
@@ -104,10 +104,6 @@
 #	endif
 #endif
 
-#ifndef CRC32_POLY_LE
-#define CRC32_POLY_LE 0xedb88320
-#endif
-
 /*
  * Allocate memory for LZMA2 decoder. xz_dec_lzma2_reset() must be used
  * before calling xz_dec_lzma2_run().
_
